runtime-sdk/modules/rofl-market/src/policy.rs

```diff
@@ -0,0 +1,123 @@
+use std::collections::BTreeMap;
+
+use base64::prelude::*;
+
+use oasis_runtime_sdk::{
+    core::{
+        common::crypto::signature::{PublicKey, Signature},
+        consensus::registry::{EndorsedCapabilityTEE, Node},
+        host::attestation::{LabelAttestation, ATTEST_LABELS_SIGNATURE_CONTEXT},
+    },
+    modules::rofl::{
+        policy::{AllowedEndorsement, EndorsementPolicyEvaluator},
+        Error,
+    },
+    types::address::Address,
+    Context,
+};
+
+use crate::{state, types::InstanceId};
+
+/// Name of the ROFL app instance metadata key used to store the provider attestation.
+pub const METADATA_KEY_POLICY_PROVIDER_ATTESTATION: &str = "net.oasis.policy.provider";
+/// Name of the provider label set by the scheduler.
+pub const LABEL_PROVIDER: &str = "net.oasis.provider";
+
+/// Value of the `LABEL_PROVIDER` label as set by the scheduler.
+#[derive(Clone, Debug, Default, cbor::Encode, cbor::Decode)]
+pub struct ProviderLabel {
+    /// Address of the provider.
+    pub provider: Address,
+    /// Instance identifier.
+    pub instance: InstanceId,
+}
+
+/// Provider attestation metadata stored in `METADATA_KEY_POLICY_PROVIDER_ATTESTATION` label.
+#[derive(Clone, Debug, Default, cbor::Encode, cbor::Decode)]
+pub struct ProviderAttestation {
+    /// A CBOR-serialized `LabelAttestation`.
```

Comment:
In other structs you used name `attestation` and comment `CBOR-serialized label attestation.`.